<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge">

  <title>Darya Vanichkina | Mixomics PCA</title>
  <meta name="description" content="Darya Vanichkina's homepage.
">

  <link rel="shortcut icon" href="http://localhost:4000/assets/img/favicon.ico">

  <link rel="stylesheet" href="http://localhost:4000/assets/css/main.css">
  <link rel="canonical" href="http://localhost:4000/wikipages/r/mixomics/">
</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    
    <span class="site-title">
        
        <strong>Darya</strong> Vanichkina
    </span>
    

    <nav class="site-nav">
      <input type="checkbox" id="nav-trigger" class="nav-trigger" />
        <label for="nav-trigger">
          <span class="menu-icon">
            <svg viewBox="0 0 18 15" width="18px" height="15px">
              <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
              <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
              <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
            </svg>
          </span>
        </label>

      <div class="trigger">
        <!-- About -->
        <a class="page-link" href="http://localhost:4000/">home</a>

        <!-- Blog -->
        <a class="page-link" href="http://localhost:4000/blog/">blog</a>

        <!-- Pages -->
        
          
            <a class="page-link" href="http://localhost:4000/404.html">404: Page not found</a>
          
        
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/biomart/">biomaRt</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/biomartseq/">Sequence with biomaRt</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/python/chromosomesize/">Make chromosome size file from reference fasta</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/contacts">contact</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/ct/">Pretty print to one column (equivalent of ls -1)</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/ngs/dgea/">Notes on differential gene expression analysis</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/gitcontribute/">Contributing to other repositories</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/gitlog/">Customize how git log is displayed</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/grep/">grep</a>
          
        
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/junctions/">Simple awk scripts that extracts junctions from SAM</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/latex/">LaTeX notes</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/least1valuerow/">At least one value in row</a>
          
        
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/ngs/mats/">MATS</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/merge2df/">Merge more than 2 data frames</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/mixomics/">Mixomics PCA</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/modenotin/">Mode and not in</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/python/openwritelinebyline/">Open file and write line by line</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/orderdf/">Order data frame</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/outlierdetectionR/">Beautiful outlier test and visualization in R</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/overlapviolin/">Creating overlapping violin plots using ggplot2</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/paste/">Paste tab-delimited text into data table in R</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/sedwnewline/">Sed with newline and quotes</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/ngs/star/">STAR</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/teaching/">teaching</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/venn/">Venn</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/volcano/">Volcano</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/python/whatvariablesarehere/">All variables defined and their values</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wiki/">Bash</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/xlsx/">Export to xlsx</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/xtable/">Pretty output with xtable</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/char2var/">Character to variable and vice versa</a>
          
        

        <!-- CV link -->
        <!-- <a class="page-link" href="http://localhost:4000/assets/pdf/CV.pdf">vitae</a> -->

      </div>
    </nav>

  </div>

</header>



    <div class="page-content">
      <div class="wrapper">
        <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Compare Sam's with Illumina body map -----
IBM &lt;- read.table("~/Documents/A_A_MyProjects/01_ActivityStudy/CounttablesForQuickExploration/IBM/bigIBM.table", col.names=c("Geneadipose", "adipose","Geneadrenal", "adrenal","Genebrain", "brain","Genebreast", "breast","Genecolon", "colon","Geneheart", "heart","Genekidney", "kidney","Geneliver", "liver","Genelung", "lung","Genelymphnode", "lymphnode","Geneovary", "ovary","Geneprostate", "prostate","Geneskeletalmuscle", "skeletalmuscle","Genetestes", "testes","Genethyroid", "thyroid","Gene", "whiteblood"), sep="\t")
IBM$Geneadipose &lt;- NULL
IBM$Geneadrenal &lt;- NULL
IBM$Genebrain &lt;- NULL
IBM$Genebreast &lt;- NULL
IBM$Genecolon &lt;- NULL
IBM$Geneheart &lt;- NULL
IBM$Genekidney &lt;- NULL
IBM$Geneliver &lt;- NULL
IBM$Genelung &lt;- NULL
IBM$Genelymphnode &lt;- NULL
IBM$Geneovary &lt;- NULL
IBM$Geneprostate &lt;- NULL
IBM$Geneskeletalmuscle &lt;- NULL
IBM$Genetestes &lt;- NULL
IBM$Genethyroid &lt;- NULL
row.names(IBM) &lt;- IBM$Gene
#IBM$Gene &lt;- NULL
counttable.sam &lt;- backup.sam
IBMUs &lt;- merge.data.frame(IBM, counttable.sam, by.x="Gene", by.y="Gene")
tmp &lt;- merge.data.frame(IBMUs, counttable.C32,by.x="Gene", by.y="Gene" )
IBMUs &lt;- tmp
rm(IBM, tmp)
							
conds &lt;- factor(c( "adipose", "adrenal", "brain", "breast", "colon", "heart", "kidney", "liver", "lung", "lymphnode", "ovary", "prostate", "skeletalmuscle", "testes", "thyroid", "whiteblood","C11d0", "C11d0", "C11d0", "MC3d0", "MC3d0", "MC3d0","C11d34", "C11d34", "C11d34", "MC3d34", "MC3d34", "MC3d34", "C32d42", "C32d42", "C32d42"))
row.names(IBMUs) &lt;- IBMUs$Gene
IBMUs$Gene &lt;- NULL													
d &lt;- newCountDataSet(IBMUs, conds)
d &lt;- estimateSizeFactors(d)
d &lt;- estimateDispersions(d,method = "blind")
vsdFull = varianceStabilizingTransformation( d )
library("RColorBrewer")
library("gplots")
# heatmap of top 10000
select = order(rowMeans(counts(d)), decreasing=TRUE)[1:10000]
hmcol = colorRampPalette(brewer.pal(9, "GnBu"))(100)
heatmap.2(exprs(vsdFull)[select,], col = hmcol, trace="none", margin=c(10, 6))


# clustering by distance
dists = dist( t( exprs(vsdFull) ) )
mat = as.matrix( dists )
rownames(mat) = colnames(mat) = with(pData(d), paste(condition))
heatmap.2(mat, trace="none", col = rev(hmcol), margin=c(13, 13))
# PCA
library(mixOmics)
forpca &lt;- as.data.frame(t(exprs(vsdFull)))
dim(forpca)
# [1]    31 46734
pcatune(forpca, center = TRUE, scale. = FALSE)
# 29 percent is explained by dim1, 9 for dim2, 7 for dim3
# remove the columns (genes) for which the variance is the same
forpca2 &lt;- forpca[,apply(forpca, 2, var, na.rm=TRUE) != 0]
# the proportions of explained variance do no change
result &lt;- pca(forpca2, ncomp = 3, center = TRUE, scale. = TRUE)
# Make it pretty
## Now lets add colours to each type of cell line
tmp1 &lt;- as.data.frame(c( "adipose", "adrenal", "brain", "breast", "colon", "heart", "kidney", "liver", "lung", "lymphnode", "ovary", "prostate", "skeletalmuscle", "testes", "thyroid", "whiteblood", "C11d0", "C11d0", "C11d0", "C11d34", "C11d34", "C11d34", "MC3d0", "MC3d0", "MC3d0", "MC3d34", "MC3d34", "MC3d34", "C32d42", "C32d42", "C32d42" ))
names(tmp1) &lt;- c("class")
# tmp &lt;- cbind(tmp1,forpca2)


col.drug = as.numeric(as.factor(tmp1$class))
col.drug[col.drug == 1] &lt;- 'antiquewhite'
col.drug[col.drug == 2] &lt;- 'mediumaquamarine'
col.drug[col.drug == 3] &lt;- 'magenta1'
col.drug[col.drug == 4] &lt;- 'azure4'
col.drug[col.drug == 5] &lt;- 'blue'
col.drug[col.drug == 6] &lt;- 'blueviolet'
col.drug[col.drug == 7] &lt;- 'brown'
col.drug[col.drug == 8] &lt;- 'aquamarine'
col.drug[col.drug == 9] &lt;- 'chartreuse'
col.drug[col.drug == 10] &lt;- 'darkgreen'
col.drug[col.drug == 11] &lt;- 'cornflowerblue'
col.drug[col.drug == 12] &lt;- 'cyan'
col.drug[col.drug == 13] &lt;- 'darkblue'
col.drug[col.drug == 14] &lt;- 'black'
col.drug[col.drug == 15] &lt;- 'deeppink'
col.drug[col.drug == 16] &lt;- 'lightpink'
col.drug[col.drug == 17] &lt;- 'yellow'
col.drug[col.drug == 18] &lt;- 'navy'
col.drug[col.drug == 19] &lt;- 'red'
col.drug[col.drug == 20] &lt;- 'darkolivegreen'
col.drug[col.drug == 21] &lt;- 'darkorchid4'

plotIndiv(result, comp = 1:2, ind.names = FALSE, col = col.drug, pch = 16)
legend(20, 270, unique(tmp1$class), col = unique(col.drug), pch = 16, title = "Dataset", ncol = 3, cex = 0.6)

# plot3dIndiv(result, comp = 1:3, ind.names = FALSE, col = col.drug, cex=6, axes.box = "both")
# rgl.postscript('cs_multi_plotIndiv_final.ps')
# Clean-up
rm(mat, d, dists, hmcol, select, vsdFull)
</code></pre></div></div>

      </div>
    </div>

    <footer>

  <div class="wrapper">
    &copy; Copyright 2018 Darya Vanichkina.
    Powered by <a href="http://jekyllrb.com/" target="_blank">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio">al-folio</a> theme.

    
  </div>

</footer>


    <!-- Load jQuery -->
<script src="//code.jquery.com/jquery-1.12.4.min.js"></script>

<!-- Load Common JS -->
<script src="http://localhost:4000/assets/js/common.js"></script>


<!-- Load KaTeX -->
<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/KaTeX/0.9.0/katex.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/KaTeX/0.9.0/katex.min.js"></script>
<script src="http://localhost:4000/assets/js/katex.js"></script>




<!-- Include custom icon fonts -->
<link rel="stylesheet" href="http://localhost:4000/assets/css/fontawesome-all.min.css">
<link rel="stylesheet" href="http://localhost:4000/assets/css/academicons.min.css">

<!-- Google Analytics -->
<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','//www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-XXXXXXXXX', 'auto');
ga('send', 'pageview');
</script>


  </body>

</html>
