<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge">

  <title>Darya Vanichkina | Sequence with biomaRt</title>
  <meta name="description" content="Darya Vanichkina's homepage.
">

  <link rel="shortcut icon" href="http://localhost:4000/assets/img/favicon.ico">

  <link rel="stylesheet" href="http://localhost:4000/assets/css/main.css">
  <link rel="canonical" href="http://localhost:4000/wikipages/r/biomartseq/">
</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    
    <span class="site-title">
        
        <strong>Darya</strong> Vanichkina
    </span>
    

    <nav class="site-nav">
      <input type="checkbox" id="nav-trigger" class="nav-trigger" />
        <label for="nav-trigger">
          <span class="menu-icon">
            <svg viewBox="0 0 18 15" width="18px" height="15px">
              <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
              <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
              <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
            </svg>
          </span>
        </label>

      <div class="trigger">
        <!-- About -->
        <a class="page-link" href="http://localhost:4000/">home</a>

        <!-- Blog -->
        <a class="page-link" href="http://localhost:4000/blog/">blog</a>

        <!-- Pages -->
        
          
            <a class="page-link" href="http://localhost:4000/404.html">404: Page not found</a>
          
        
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/biomart/">biomaRt</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/biomartseq/">Sequence with biomaRt</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/python/chromosomesize/">Make chromosome size file from reference fasta</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/contacts">contact</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/ct/">Pretty print to one column (equivalent of ls -1)</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/ngs/dgea/">Notes on differential gene expression analysis</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/gitcontribute/">Contributing to other repositories</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/gitlog/">Customize how git log is displayed</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/grep/">grep</a>
          
        
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/junctions/">Simple awk scripts that extracts junctions from SAM</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/latex/">LaTeX notes</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/least1valuerow/">At least one value in row</a>
          
        
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/ngs/mats/">MATS</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/merge2df/">Merge more than 2 data frames</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/mixomics/">Mixomics PCA</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/modenotin/">Mode and not in</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/python/openwritelinebyline/">Open file and write line by line</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/orderdf/">Order data frame</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/outlierdetectionR/">Beautiful outlier test and visualization in R</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/overlapviolin/">Creating overlapping violin plots using ggplot2</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/paste/">Paste tab-delimited text into data table in R</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/_pages/publications/">publications</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/sedwnewline/">Sed with newline and quotes</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/ngs/star/">STAR</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/teaching/">teaching</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/venn/">Venn</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/volcano/">Volcano</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/python/whatvariablesarehere/">All variables defined and their values</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wiki/">Bash</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/xlsx/">Export to xlsx</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/xtable/">Pretty output with xtable</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/char2var/">Character to variable and vice versa</a>
          
        

        <!-- CV link -->
        <!-- <a class="page-link" href="http://localhost:4000/assets/pdf/CV.pdf">vitae</a> -->

      </div>
    </nav>

  </div>

</header>



    <div class="page-content">
      <div class="wrapper">
        <h2 id="sequence-with-biomart">Sequence with biomart</h2>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>getSequence {biomaRt}	R Documentation

Retrieves sequences

Description

This function retrieves sequences given the chomosome, start and end position or a list of identifiers. Using getSequence in web service mode (default) generates 5' to 3' sequences of the requested type on the correct strand. The type of sequence returned can be specified by the seqType argument which takes the following values: 'cdna';'peptide' for protein sequences;'3utr' for 3' UTR sequences,'5utr' for 5' UTR sequences; 'gene_exon' for exon sequences only; 'transcript_exon_intron' gives the full unspliced transcript, that is exons + introns;'gene_exon_intron' gives the exons + introns of a gene;'coding' gives the coding sequence only;'coding_transcript_flank' gives the flanking region of the transcript including the UTRs, this must be accompanied with a given value for the upstream or downstream attribute;'coding_gene_flank' gives the flanking region of the gene including the UTRs, this must be accompanied with a given value for the upstream or downstream attribute; 'transcript_flank' gives the flanking region of the transcript exculding the UTRs, this must be accompanied with a given value for the upstream or downstream attribute; 'gene_flank' gives the flanking region of the gene excluding the UTRs, this must be accompanied with a given value for the upstream or downstream attribute. In MySQL mode the getSequence function is more limited and the sequence that is returned is the 5' to 3'+ strand of the genomic sequence, given a chromosome, as start and an end position. So if the sequence of interest is the minus strand, one has to compute the reverse complement of the retrieved sequence, which can be done using functions provided in the matchprobes package. The biomaRt vignette contains more examples on how to use this function.




Usage

getSequence( chromosome, start, end, id, type, seqType, upstream, downstream, mart, verbose=FALSE)
Arguments

chromosome	 Chromosome name
start	 start position of sequence on chromosome
end	 end position of sequence on chromosome
id	 An identifier or vector of identifiers.
type	 The type of identifier used. Supported types are hugo, ensembl, embl, entrezgene, refseq, ensemblTrans and unigene. Alternatively one can also use a filter to specify the type. Possible filters are given by the listFilters function
seqType	 Type of sequence that you want to retrieve. Allowed seqTypes are: cdna, peptide, 3utr, 5utr, genomic
upstream	 To add the upstream sequence of a specified number of basepairs to the output.
downstream	 To add the downstream sequence of a specified number of basepairs to the output.
mart	 object of class Mart created using the useMart function
verbose	 If verbose = TRUE then the XML query that was send to the webservice will be displayed.
</code></pre></div></div>

<p>Examples</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>if(interactive()){
mart &lt;- useMart("ensembl",dataset="hsapiens_gene_ensembl")

seq = getSequence(id="BRCA1", type="hgnc_symbol", seqType="peptide", mart = mart)
show(seq)

seq = getSequence(id="1939_at", type="affy_hg_u95av2", seqType="gene_flank",upstream = 20, mart = mart)
show(seq)

}

seq = getSequence(id=U1all$ensembl_gene_id, type="ensembl_gene_id", seqType="cdna", mart = ensembl)
write.fasta(sequences=as.list(seq$cdna),names=seq$ensembl_gene_id,file.out="write.my.dna.fasta")
</code></pre></div></div>

      </div>
    </div>

    <footer>

  <div class="wrapper">
    &copy; Copyright 2018 Darya Vanichkina.
    Powered by <a href="http://jekyllrb.com/" target="_blank">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio">al-folio</a> theme.

    
  </div>

</footer>


    <!-- Load jQuery -->
<script src="//code.jquery.com/jquery-1.12.4.min.js"></script>

<!-- Load Common JS -->
<script src="http://localhost:4000/assets/js/common.js"></script>


<!-- Load KaTeX -->
<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/KaTeX/0.9.0/katex.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/KaTeX/0.9.0/katex.min.js"></script>
<script src="http://localhost:4000/assets/js/katex.js"></script>




<!-- Include custom icon fonts -->
<link rel="stylesheet" href="http://localhost:4000/assets/css/fontawesome-all.min.css">
<link rel="stylesheet" href="http://localhost:4000/assets/css/academicons.min.css">

<!-- Google Analytics -->
<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','//www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-XXXXXXXXX', 'auto');
ga('send', 'pageview');
</script>


  </body>

</html>
