<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge">

  <title>Darya Vanichkina | Notes on differential gene expression analysis</title>
  <meta name="description" content="Darya Vanichkina's homepage.
">

  <link rel="shortcut icon" href="http://localhost:4000/assets/img/favicon.ico">

  <link rel="stylesheet" href="http://localhost:4000/assets/css/main.css">
  <link rel="canonical" href="http://localhost:4000/wikipages/ngs/dgea/">
</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    
    <span class="site-title">
        
        <strong>Darya</strong> Vanichkina
    </span>
    

    <nav class="site-nav">
      <input type="checkbox" id="nav-trigger" class="nav-trigger" />
        <label for="nav-trigger">
          <span class="menu-icon">
            <svg viewBox="0 0 18 15" width="18px" height="15px">
              <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
              <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
              <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
            </svg>
          </span>
        </label>

      <div class="trigger">
        <!-- About -->
        <a class="page-link" href="http://localhost:4000/">home</a>

        <!-- Blog -->
        <a class="page-link" href="http://localhost:4000/blog/">blog</a>

        <!-- Pages -->
        
          
            <a class="page-link" href="http://localhost:4000/404.html">404: Page not found</a>
          
        
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/biomart/">biomaRt</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/biomartseq/">Sequence with biomaRt</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/python/chromosomesize/">Make chromosome size file from reference fasta</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/contacts">contact</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/ct/">Pretty print to one column (equivalent of ls -1)</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/ngs/dgea/">Notes on differential gene expression analysis</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/gitcontribute/">Contributing to other repositories</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/gitlog/">Customize how git log is displayed</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/grep/">grep</a>
          
        
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/junctions/">Simple awk scripts that extracts junctions from SAM</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/latex/">LaTeX notes</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/least1valuerow/">At least one value in row</a>
          
        
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/ngs/mats/">MATS</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/merge2df/">Merge more than 2 data frames</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/mixomics/">Mixomics PCA</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/modenotin/">Mode and not in</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/python/openwritelinebyline/">Open file and write line by line</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/orderdf/">Order data frame</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/outlierdetectionR/">Beautiful outlier test and visualization in R</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/overlapviolin/">Creating overlapping violin plots using ggplot2</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/paste/">Paste tab-delimited text into data table in R</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/sedwnewline/">Sed with newline and quotes</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/ngs/star/">STAR</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/teaching/">teaching</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/venn/">Venn</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/volcano/">Volcano</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/python/whatvariablesarehere/">All variables defined and their values</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wiki/">Bash</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/xlsx/">Export to xlsx</a>
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/xtable/">Pretty output with xtable</a>
          
        
          
        
          
            <a class="page-link" href="http://localhost:4000/wikipages/r/char2var/">Character to variable and vice versa</a>
          
        

        <!-- CV link -->
        <!-- <a class="page-link" href="http://localhost:4000/assets/pdf/CV.pdf">vitae</a> -->

      </div>
    </nav>

  </div>

</header>



    <div class="page-content">
      <div class="wrapper">
        <h2 id="voom-to-rpkm">Voom to RPKM</h2>
<p>( from the Bioconductor mailing list; Gordon Smythe)</p>

<p>voom() outputs normalized log-cpm values.  Just plot them.
Ex.:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>v &lt;- voom(y, design)
</code></pre></div></div>

<p>the log-cpm values are in v$E. To get RPKM:</p>

<p>Take log2CPM values from voom, and subtract log2(GeneLength/1000) from each row, where GeneLength is the effective length of each gene in bases.  This will convert the log2CPM values into log2 FPKM.</p>

<h2 id="testing-for-no-change-in-rna-seq-differential-gene-expression">Testing for no change in RNA-seq differential gene expression</h2>
<p>( from the Bioconductor mailing list; Simon Anders)</p>

<p>DESeq2  provides a empirical- Bayes style shrinkage estimates of coefficients (i.e., log fold changes) and also estimates standard errors for these coefficients (taken from the the reciprocal curvature of the posterior). Your task is one of the application we had in mind for this.You want to know whether the fold change is zero or nearly zero, and you also want to ascertain that this estimate of a nearly-zero fold change is a precise one. So, to find genes whose absolute log fold change is <em>reliably</em> smaller than some threshold, take all those genes for which the estimated abs log fold change is below the threshold and the standard error is well below the threshold, too.The reason, why I write “below a threshold” rather than “equal to zero”is that it’s biologically unreasonable to assume that a treatment has exactly zero effect on a gene’s expression. Gene regulation is such an interconnected network that, at least in my view, every gene will react ever so slightly to every perturbance, but often, this reaction is too small to be measurable. This is why you should define a threshold for”biologically unlikely to be relevant”. Let’s say, we don’t believe that an expression change of less than 15% (0.2 on a log2 scale) is worth bothering. So, if you might take all genes with abs log2 fold change below 0.2, and furthermore require that the standard error of this log2fold change estimate is below, say, 0.05, than you will get gene with true values well below at least 0.3 or so.In the end, the thresholds won’t matter too much but if you want real p values, this is possible, too: Simply divide the log2 fold change estimates by their standard errors to get z scores (this works because the sampling distribution of our fold change estimates is reasonably close to normal), and do two one-sided tests (TOST) as suggested by Ryan above.</p>

      </div>
    </div>

    <footer>

  <div class="wrapper">
    &copy; Copyright 2018 Darya Vanichkina.
    Powered by <a href="http://jekyllrb.com/" target="_blank">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio">al-folio</a> theme.

    
  </div>

</footer>


    <!-- Load jQuery -->
<script src="//code.jquery.com/jquery-1.12.4.min.js"></script>

<!-- Load Common JS -->
<script src="http://localhost:4000/assets/js/common.js"></script>


<!-- Load KaTeX -->
<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/KaTeX/0.9.0/katex.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/KaTeX/0.9.0/katex.min.js"></script>
<script src="http://localhost:4000/assets/js/katex.js"></script>




<!-- Include custom icon fonts -->
<link rel="stylesheet" href="http://localhost:4000/assets/css/fontawesome-all.min.css">
<link rel="stylesheet" href="http://localhost:4000/assets/css/academicons.min.css">

<!-- Google Analytics -->
<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','//www.google-analytics.com/analytics.js','ga');

ga('create', 'UA-XXXXXXXXX', 'auto');
ga('send', 'pageview');
</script>


  </body>

</html>
